sympy/physics/mechanics/system.py

```diff
@@ -1,8 +1,1009 @@
+    >>> system.loads
+    ((rail_masscenter, - g*rail_mass*rail_frame.y), (cart_masscenter, - cart_mass*g*rail_frame.y), (bob_masscenter, - bob_mass*g*rail_frame.y), (cart_masscenter, F*rail_frame.x))
+
+    With the entire system being defined, we can now form the equations of
+    motion. Before forming the equations of motion, one can also run some checks
+    to check for possible errors.
+
+    >>> system.validate_system()
+    >>> system.form_eoms()
+    Matrix([
+    [bob_mass*l*u_pin**2*sin(q_pin) - bob_mass*l*cos(q_pin)*u_pin' - (bob_mass + cart_mass)*u_slider' + F],
+    [                   -bob_mass*g*l*sin(q_pin) - bob_mass*l**2*u_pin' - bob_mass*l*cos(q_pin)*u_slider']])
+    >>> simplify(system.mass_matrix)
+    Matrix([
+    [ bob_mass + cart_mass, bob_mass*l*cos(q_pin)],
+    [bob_mass*l*cos(q_pin),         bob_mass*l**2]])
+    >>> system.forcing
+    Matrix([
+    [bob_mass*l*u_pin**2*sin(q_pin) + F],
+    [          -bob_mass*g*l*sin(q_pin)]])
+
+    The complexity of the above example can be increased if we add a holonomic
+    constraint, to prevent the particle from moving in the horizontal (x)
+    direction. This can be done by adding a holonomic constraint. After, which
+    we should also redefine what our (in)dependent generalized coordinates and
+    speeds are. Note that the bakcend for forming the equations of motion is
+    resetted automatically, because the system properties are changed in this
+    process.
+
+    >>> type(system.eom_method)
+    <class 'sympy.physics.mechanics.kane.KanesMethod'>
+    >>> system.add_holonomic_constraints(
+    ...     bob.masscenter.pos_from(rail.masscenter).dot(system.x)
+    ... )
+    >>> system.q_ind = system.get_joint('pin').coordinates
+    >>> system.q_dep = system.get_joint('slider').coordinates
+    >>> system.u_ind = system.get_joint('pin').speeds
+    >>> system.u_dep = system.get_joint('slider').speeds
+    >>> type(system.eom_method)
+    <class 'NoneType'>
+
+    With the updated system the equations of motion can be formed again.
+
+    >>> system.validate_system()
+    >>> system.form_eoms()
+    Matrix([[-bob_mass*g*l*sin(q_pin) - bob_mass*l**2*u_pin' - bob_mass*l*cos(q_pin)*u_slider' - l*(bob_mass*l*u_pin**2*sin(q_pin) - bob_mass*l*cos(q_pin)*u_pin' - (bob_mass + cart_mass)*u_slider')*cos(q_pin) - l*F*cos(q_pin)]])
+    >>> simplify(system.mass_matrix)
+    Matrix([
+    [bob_mass*l**2*sin(q_pin)**2, -cart_mass*l*cos(q_pin)],
+    [               l*cos(q_pin),                       1]])
+    >>> simplify(system.forcing)
+    Matrix([
+    [-l*(bob_mass*g*sin(q_pin) + bob_mass*l*u_pin**2*sin(2*q_pin)/2 + F*cos(q_pin))],
+    [                                                         l*u_pin**2*sin(q_pin)]])
+
+    """
+
+    def __init__(self, origin=None, frame=None):
+        if origin is None:
+            origin = Point('inertial_origin')
+        elif not isinstance(origin, Point):
+            raise TypeError('Origin must be and instance of Point.')
```

Comment:
```suggestion
            raise TypeError('Origin must be an instance of Point.')
```
Or "Origin must be a Point"